CONTRIBUTING.md

```diff
@@ -60,6 +60,7 @@ But the following guidelines provide a more detailed set of requirements that we
 2. The Pull Request *should* contain a link to the JIRA issue(s) at the start of the PR description (only minor changes to script/text files are exempt from this rule). If the engineer wishes to address multiple issues and they are closely related then they can be addressed in a single PR. The JIRA must contain sufficient information to enable the reader to understand what the issue is, so at a minimum the description field of the JIRA must be present and legible/clear.
 3. Engineers are not allowed to submit PRs which only contain formatting changes. The guidance on formatting code are covered in the [Coding Guidelines](#coding-guidelines) section below.
 4. New PRs are tested against multiple Jenkins CI axes. If you know that a change only affects particular axes then you can disable the ones that aren't required, ask a team member for clarification if you're not sure (or look at how the `PROFILE` variable is used in the CI script `scripts/hudson/narayana.sh`. When you first create a PR, the PR description field contains basic instructions about how to disable a CI test axis.
+5. If the pull request is not ready to be reviewed yet, you must add the label "Hold" - this can be added when the pull request is being created.
```

Comment:
Can we also add something to deter users from using our CI to test their code changes as a way of avoiding their responsibility of doing the testing locally first.

Reply:
Yes please! +1 from me!

Reply:
I would prefer to deal with that separate to this pull request which just raises the attention on the Hold label and so I expect it to be a different decision.

If the project doesn't want people to use CI for pre-testing we could make the poller configuration ignore pull requests which have the "Hold" label and describe that restriction in this CONTRIBUTING.md. If that is what the project would like I still think that could be follow up once it's agreed by the project.

Reply:
Personally I find the Ci testing useful because I find running all profiles locally is quite time-consuming. I agree the core profile should be run locally first then. Just to mention the CI poller is ignoring all pull request at the moment (https://github.com/jbosstm/narayana/pull/2085)

Reply:
What about if we ask users to check that github actions (LRA github action https://github.com/jbosstm/narayana/blob/main/.github/workflows/lra-test.yml) pass on their branch before creating the PR?

Reply:
BTW I agree with Tom this can be done in a separate PR.

Reply:
> I would prefer to deal with that separate to this pull request (which just raises the attention on the Hold label) - I expect it to be a different decision.
> 
> If the project doesn't want people to use CI for pre-testing we could make the poller configuration ignore pull requests which have the "Hold" label and describe that restriction in this CONTRIBUTING.md. If that is what the project would like I still think that could be follow up once it's agreed by the project.

Well it gives the wrong message, if I read that I would infer that the team is okay with people using our CI as a testbed for their development. And if you are telling them to use the Hold label wisely then using the label just to get some free testing seems like an unwise use of the label. So telling them under what circumstances they can use the Hold label needs to include our caveat, ie the caveat belongs with the statement you are proposing in this PR.

Reply:
I like the idea of the GitHub action from @marcosgopen but I think we should add a CORE version rather than use the LRA one personally.

Reply:
@jmfinelli @mmusgrov please can you update your reviews to request changes if you would like the pull request changing?

Reply:
> Personally I find the Ci testing useful because I find running all profiles locally is quite time-consuming. I agree the core profile should be run locally first then. Just to mention the CI poller is ignoring all pull requests at the moment (#2085)

I would like to mention that we don't have unlimited resources on our CI. There is a maximum number of agents that can run in parallel. In other words, if contributors use our CI to do the initial testing of their PRs, our CI has to deal with a workload that may be over the allocated resources. The reason we use this limitation in our CI comes from the assumption that users raise PRs only when they know (with a fair level of confidence) that their PR is ready to be merged.

Moreover, although this might not be everyone's concern, using our CI to run the first round of testing for a new PR is not a green option. What I mean is that testing on our CI still means  that there is a CPU running somewhere, which means that we're wasting energy (and money). Being conscious about how we use our CI is another way to be conscious about the environment.

Reply:
So @tomjenkinson something like the following could be used:

```
5. Although the expectation is that the engineer raising the PR has tested his changes prior to submitting the request to merge changes, there are some circumstances where this may not be possible in which case you must add the label "Hold" and update the PR description indicating why it isn't ready for review just yet. This policy is a signal to reviewers that the changes are not yet ready to be reviewed.
```

Reply:
Regarding the CORE github action see https://github.com/jbosstm/narayana/pull/2124